base/namedtuple.jl

```diff
@@ -113,6 +113,15 @@ function convert(::Type{NamedTuple{names,T}}, nt::NamedTuple{names}) where {name
     NamedTuple{names,T}(T(nt))
 end
 
+function Tuple(nt::NamedTuple{names}) where {names}
+    if @generated
+        return Expr(:tuple, Any[:(getfield(nt, $(QuoteNode(n)))) for n in names]...)
+    else
+        return tuple(nt...)
+    end
+end
+(::Type{T})(nt::NamedTuple{names}) where {names, T <: Tuple} = convert(T, Tuple(nt))
```

Comment:
`names` is now unnecessary here.

Reply:
This is because I have an irrational fear of the method specificity rules and resultant ambiguities (w.r.t. the method above which does specialize on `names`). I'll get rid of it. :)